src/preproc/preprocessing_pass_api.cpp

```diff
@@ -0,0 +1,26 @@
+/*********************                                                        */
+/*! \file preprocessing_pass_api.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **  Justin Xu
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Implementation for preprocessing pass api for passes
+ **
+ ** Implementation for preprocessing pass api for passes. Takes in variables and
+ *methods from d_smt
+ *and d_private and allows for access to the passes.
```

Comment:
Comment formatting is off, the line should start with two stars (I think clang-format did this when it was wrapping the long lines). There is also no `d_private` variable right now. Maybe say something like: "Implementation of the preprocessing API for passes. Stores a pointer to the SmtEngine that it was created in and exposes parts of the solver to the preprocessing passes."